File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_tile_clause.c
#include <stdio.h>
#include <stdlib.h>
#include <openacc.h>

#define N 100
#define TILE_SIZE 10

int main() {
    int a[N][N], b[N][N];
    int i, j, tile_i, tile_j;

    // Initialize the arrays
    for (i = 0; i < N; i++) {
        for (j = 0; j < N; j++) {
            a[i][j] = i + j;
            b[i][j] = 0;
        }
    }

    // Tile the loop
    #pragma acc data copyin(a) copyout(b)
    {
        #pragma acc parallel loop tile(TILE_SIZE,TILE_SIZE)
        for (tile_i = 0; tile_i < N; tile_i += TILE_SIZE) {
            for (tile_j = 0; tile_j < N; tile_j += TILE_SIZE) {
                #pragma acc loop
                for (i = tile_i; i < tile_i + TILE_SIZE; i++) {
                    for (j = tile_j; j < tile_j + TILE_SIZE; j++) {
                        b[i][j] = a[i][j];
                    }
                }
            }
        }
    }

    // Check the result
    for (i = 0; i < N; i++) {
        for (j = 0; j < N; j++) {
            if (a[i][j] != b[i][j]) {
                printf("Error: a[%d][%d] = %d, b[%d][%d] = %d\n", i, j, a[i][j], i, j, b[i][j]);
                return 1;
            }
        }
    }

    printf("Test passed.\n");
    return 0;
}

File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_tile_clause.f90
!T1:loop construct tile clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER, PARAMETER :: N = 100
        INTEGER, PARAMETER :: TILE_SIZE = 10
        INTEGER :: i, j, sum1, sum2
        INTEGER :: errors = 0
        INTEGER, DIMENSION(N,N) :: a

        ! Initialize array
        DO j = 1, N
          DO i = 1, N
            a(i,j) = i + j
          END DO
        END DO

        ! Sum before tile clause
        sum1 = 0
        DO j = 1, N
          DO i = 1, N
            sum1 = sum1 + a(i,j)
          END DO
        END DO

        ! Apply tile clause
        !$acc data copy(a)
        !$acc parallel loop tile(TILE_SIZE,TILE_SIZE)
        DO j = 1, N
          DO i = 1, N
            a(i,j) = a(i,j) * 2
          END DO
        END DO
        !$acc end parallel loop
        !$acc end data

        ! Sum after tile clause
        sum2 = 0
        DO j = 1, N
          DO i = 1, N
            sum2 = sum2 + a(i,j)
          END DO
        END DO

        ! Check if sums are equal
        IF (sum1 .ne. sum2) THEN
          errors = errors + 1
        END IF

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM